### ISSUE TITLE
Interpreter AllocateTensors() does not finish or report error

### ISSUE THREAD

**Issue by PMatthaei** [tensorflow/tensorflow#43238]
@tensorflow/micro

**System information**
- Host OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows
- TensorFlow installed from (source or binary): Source - Arduino TensorFlowLite version 2.1.0-ALPHA(not precompiled)
- Tensorflow version (commit SHA if source):
- Target platform (e.g. Arm Mbed OS, Arduino Nano 33 etc.): SparkFun Edge

**Describe the problem**

Allocating a model does not finish either with a error or success. Before I added the SOFTMAX and RESHAPE ops I received at least messages through the report serial at 9600. Now `Serial.print("Allocating Tensors ...\n");` is printing and nothing happens as if it `TfLiteStatus allocate_status = interpreter->AllocateTensors();` is stuck or an error message is not arriving. I use the biggest arena size I got via decreasing until the SRAM overflow error disappeared. The used model is included at the end. It was created from the [quantized version of mobilenet v1](http://download.tensorflow.org/models/mobilenet_v1_2018_08_02/mobilenet_v1_0.25_128_quant.tgz) via the notebook [here](https://colab.research.google.com/github/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/examples/hello_world/train/train_hello_world_model.ipynb#scrollTo=HPSFmDL7pv2L).

I am new to microcontrollers and ML.

**Please provide the exact sequence of commands/steps when you ran into the problem**
```
namespace {
	tflite::ErrorReporter* error_reporter = nullptr;
	const tflite::Model* model = nullptr;
	tflite::MicroInterpreter* interpreter = nullptr;
	TfLiteTensor* input = nullptr;

	// An area of memory to use for input, output, and intermediate arrays.
	constexpr int kTensorArenaSize = 270 * 1024;
	static uint8_t tensor_arena[kTensorArenaSize];
} 

void setup() {

	Serial.begin(460800);
	do {
		delay(500);
	} while (!Serial);

	static tflite::MicroErrorReporter micro_error_reporter;
	error_reporter = &micro_error_reporter;

	Serial.print("Tensorflow setup...\n");

	Serial.print("Loading model ...\n");

	model = tflite::GetModel(g_object_detect_model_data);
	if (model->version() != TFLITE_SCHEMA_VERSION) {
		Serial.printf("Model provided is schema version %d not equal to supported version %d.",
			model->version(), TFLITE_SCHEMA_VERSION);
		return;
	}

	Serial.print("Model loaded ...\n");

	Serial.print("Building interpreter ...\n");

	static tflite::MicroMutableOpResolver<5> micro_op_resolver;

	micro_op_resolver.AddBuiltin(
		tflite::BuiltinOperator_DEPTHWISE_CONV_2D,
		tflite::ops::micro::Register_DEPTHWISE_CONV_2D());

	micro_op_resolver.AddBuiltin(tflite::BuiltinOperator_CONV_2D,
		tflite::ops::micro::Register_CONV_2D());

	micro_op_resolver.AddBuiltin(tflite::BuiltinOperator_AVERAGE_POOL_2D,
		tflite::ops::micro::Register_AVERAGE_POOL_2D());

	micro_op_resolver.AddBuiltin(tflite::BuiltinOperator_SOFTMAX,
		tflite::ops::micro::Register_SOFTMAX());

	micro_op_resolver.AddBuiltin(
		tflite::BuiltinOperator_RESHAPE,
		tflite::ops::micro::Register_RESHAPE());

	static tflite::MicroInterpreter static_interpreter(
		model, micro_op_resolver, tensor_arena, kTensorArenaSize, error_reporter);
	interpreter = &static_interpreter;

	Serial.print("Interpreter built ...\n");

	Serial.print("Allocating Tensors ...\n");

	TfLiteStatus allocate_status = interpreter->AllocateTensors();

	if (allocate_status != kTfLiteOk) {
		Serial.print("AllocateTensors() failed\n");
		return;
	}

	Serial.print("Tensors allocated ...\n");

	input = interpreter->input(0);

	Serial.print("Tensorflow setup finished.\n");

}
```

[model.zip](https://github.com/tensorflow/tensorflow/files/5224464/model.zip)

**Comment by bhack** [comment]
What is you Tensorflow version?

**Comment by PMatthaei** [comment]
Added it to the description: Arduino TensorFlowLite 2.1.0-ALPHA(not precompiled) is the library I have installed via the arduino library manager.

**Comment by bhack** [comment]
Can you test with an updated version?

**Comment by PMatthaei** [comment]
![Unbenannt](https://user-images.githubusercontent.com/8292508/93205531-3fee8380-f758-11ea-804a-bb5d703acc9e.png)
These are the options given by the manager. But I can guess I can install them manually?

**Comment by bhack** [comment]
I suppose a nightly build is available in the Manager but you can still generate it yourself:
https://www.tensorflow.org/lite/microcontrollers/library#generate_the_arduino_library 
/cc @petewarden

**Comment by PMatthaei** [comment]
Following the guide you mentioned results in a tflite.zip which contains the same version as far as i can see. At least the library.properties tells so...although f.e. adding Ops to the resolver changed. But still I dont see any output after calling AllocateTensors().

**Comment by PMatthaei** [comment]
Any updates on this issue?

**Comment by sushreebarsa** [comment]
@PMatthaei  It looks like you are using an older Version of Tensorflow. Many bugs have been fixed in the latest version. Could you please execute your code using Latest  stable Version of TF 2.5 and let us know if the issue still persists? Thanks!
